crates/starknet/src/starknet/mod.rs

```diff
@@ -705,6 +706,107 @@ impl Starknet {
         transaction_to_map.get_receipt()
     }
 
+    fn get_execute_call_info(
+        execution_info: &TransactionExecutionInfo,
+        address_to_class_hash_map: &HashMap<ContractAddress, ClassHash>,
+    ) -> DevnetResult<ExecutionInvocation> {
+        Ok(match &execution_info.execute_call_info {
+            Some(call_info) => match call_info.execution.failed {
+                false => ExecutionInvocation::Succeeded(FunctionInvocation::try_from_call_info(
+                    call_info,
+                    address_to_class_hash_map,
+                )?),
+                true => {
+                    ExecutionInvocation::Reverted(starknet_types::rpc::transactions::Reversion {
+                        revert_reason: execution_info
+                            .revert_error
+                            .clone()
+                            .unwrap_or("Revert reason not found".into()),
+                    })
+                }
+            },
+            None => {
+                match execution_info.revert_error.clone() {
+                    Some(revert_reason) => ExecutionInvocation::Reverted(
+                        starknet_types::rpc::transactions::Reversion { revert_reason },
+                    ),
+                    None => {
+                        return Err(Error::UnexpectedInternalError {
+                            msg: "Simulation contains neither call_info nor revert_error".into(),
+                        });
+                    }
+                }
+            }
+        })
+    }
+
+    fn get_call_info_invocation(
+        call_info_invocation: &Option<CallInfo>,
+        address_to_class_hash_map: &HashMap<ContractAddress, ClassHash>,
+    ) -> DevnetResult<Option<FunctionInvocation>> {
```

Comment:
Wrapping an `Option` in a `(Devnet)Result` looks wrong. Also, this implementation looks like it could benefit from methods of `Option`.

Reply:
@marioiordanov? I was just refactoring your code :)

Reply:
It might just be my Rust inexperience and this pattern is considered normal, but it does look suspicious.

Reply:
> @marioiordanov? I was just refactoring your code :)

What should be my expected reply

Reply:
> What should be my expected reply

One of these:
- telling us of a magical way to refactor this not to use Results wrapped in Options
- telling us that this is a common pattern and we lack experience

Reply:
Ahh, okay. ```rust
get_call_info_invocation(call_info_invocation: &CallInfo,
        address_to_class_hash_map: &HashMap<ContractAddress, ClassHash>)-> DevnetResult<FunctionInvocation>``` . By removing the option from call_info_invocation it is removed from the result.
        
 Before executing get_call_info_invocation, the match statement will be in the form:
 ```rust
if let Some(ref x) = transaction_to_map.execution_info.validate_call_info {
            Some(FunctionInvocation::try_from_call_info(x, address_to_class_hash_map)?)
        } else {
            None
        };
 ```

Reply:
For me it doesn't look good to be Result<Option<T>> but If there is no other way I will resolve this conversation. Also creating an issue for this refactoring doesn't make sense, because this part was refactored already in this PR.